src/features/soulkeys/useSoulKeys.ts

```diff
@@ -0,0 +1,34 @@
+import { useEffect, useState } from 'react';
+
+import { useQueryClient } from 'react-query';
+
+import { Contracts } from '../cosoul/contracts';
+
+export const useSoulKeys = ({
+  contracts,
+  address,
+  subject,
+}: {
+  contracts: Contracts;
+  address: string;
+  subject: string;
+}) => {
+  const [balance, setBalance] = useState<number | null>(null);
+
+  const queryClient = useQueryClient();
+  const refresh = () => {
+    contracts.soulKeys
+      .sharesBalance(subject, address)
+      .then(b => setBalance(b.toNumber()));
+    setTimeout(() => {
```

Comment:
seems fishy...

Reply:
this is due to lack of webhooks and stuff taking long time, not sure what else to do

Reply:
yep but its ok for now until webhook